### ISSUE TITLE
git won't commit or push my file

### ISSUE THREAD

**Issue by bdacunha** [STAT545-UBC/Discussion#120]
Hello,

I have been having some trouble to commit and push from RMarkdown to GitHub. It worked the first couple of times and I managed to push a part of my work for homework 2, but when I kept going and tried to commit and push again I couldn't do it. Git wont let me select the changes I have made in order to commit. I've tried to create a different project and copied my code to it and push it to git but it wont work. I have also tried to open new files and restarting R several times but it's not working. 
Any ideas on how to overcome this problem?

**Comment by SarahQiong** [comment]
I met the same issue... When I push ,it shows somthing related to branch is wrong(sorry I forget those words). I just tried to restart R and pull several times, I have no idea what happened, but it can work now.

**Comment by jennybc** [comment]
In both cases, we would need more information to help you. Quite possibly, we'll need to just sit in front of your computer.

But in case we can troubleshoot here, I need a really precise sequence of actions and error message.

Have you *pulled* from GitHub?

Consider re-walking through the tutorials where you establish a Git <--> GitHub connection **from the shell**. Then do the same from RStudio. There is no harm in creating test repos to check things. They can easily be deleted. This might lead to a very specific description of where things break down.

@SarahQiong You, in particular, seem to have created branch, which you do not really need to do in this course. I have seen people do this in the past, thinking they are creating a directory. It is likely you have created a branch but there is no link from local to GitHub, hence your inability to push. In the Git tab in RStudio, in the upper right-hand corner, is a drop-down menu to see your different branches. If you happen to have more than one at this point, try to use that menu to get back to you **master** branch.

Just keep doing your work and, worst case, we will straighten out your Git issues on Tuesday in class or in office hours.

**Comment by bdacunha** [comment]
Ok, so I managed to create a new file and copied my code in there and with this I was able to commit and push my homework to GitHub. However, thanks to Kieran, I realized that I have something wrong with my directory (a directory inside a directory, I think) and I cannot delete (and commit and push) the files that I created before (it won't even click to commit and the options push and pull are gone).

**Comment by jennybc** [comment]
We fixed this. For posterity, we identified two problems:

  * Multiple directory names in the path to the git repo contained whitespace. This is a [known problem](http://stat545-ubc.github.io/git66_rstudio-git-github-hell.html#rstudio-is-not-making-certain-files-available-for-stagingcommiting) with RStudio's Git client and just a generally bad idea.
  * The class repo and RStudio project was sitting inside a subdirectory within student's Google Drive. It turns out this is also a bad idea and I've just added it to the troubleshooting guide
